src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -1950,6 +1950,20 @@ void HMICapabilitiesImpl::set_ccpu_version(const std::string& ccpu_version) {
   ccpu_version_ = ccpu_version;
 }
 
+void HMICapabilitiesImpl::matches_ccpu_version(
+    const std::string& ccpu_version) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (ccpu_version_ == ccpu_version) {
+    app_mngr_.set_hmi_cooperating(true);
+    return;
+  }
+
+  set_ccpu_version(ccpu_version);
+  DeleteCachedCapabilitiesFile();
+  app_mngr_.OnSendGetCapabilitiesForInterface();
+}
+
```

Comment:
@ydementieiev I believe this method does more actions than it should: checks ccpu versions, deletes cache, sends requests. And this method has `void` return value. It means that you can't be sure what the method does in the current case and with which result it completed its work. I would like to propose to create a new one enum type, e.g. CCPU_VER_MATCH {MATCH, DISMATCH} and make it as return value for the `matches_ccpu_version` method. Also, I would like to propose to split the method to two separate methods: the first method will check a `ccpu` version and return a result, the second method will make other actions based on the returned result.

Reply:
@IGapchuk  done [faf31b](https://github.com/smartdevicelink/sdl_core/pull/3269/commits/faf31b567f45e2c8d77bec815805113a0b34ba63)